data/governance/store.go

```diff
@@ -53,14 +57,38 @@ func (st *Store) WithState(state *storage.State) *Store {
 	return st
 }
 
-func (st *Store) Get(key []byte) ([]byte, error) {
-	prefixKey := append(st.prefix, storage.StoreKey(key)...)
+func (st *Store) WithHeight(height int64) *Store {
+	st.height = height
+	return st
+}
 
+func (st *Store) Get(key string) ([]byte, error) {
+	// Get the last update height for the present height
+	// LUH is unversioned
+	luh, err := st.GetLUH()
+	if err != nil {
+		panic(errors.Wrap(err, "Unable to get Last Update Height"))
+	}
+	// Get the Options from the last update Height
+	versionedKey := storage.StoreKey(string(luh) + storage.DB_PREFIX + key)
+	prefixKey := append(st.prefix, versionedKey...)
 	return st.state.Get(prefixKey)
 }
 
-func (st *Store) Set(key []byte, value []byte) error {
-	prefixKey := append(st.prefix, storage.StoreKey(key)...)
+func (st *Store) Set(key string, value []byte) error {
+	versionedKey := storage.StoreKey(string(st.height) + storage.DB_PREFIX + key)
```

Comment:
I'm sure you probably tested this, but when you build the key with the current height you should only update the LUH once the state has been comited.

Reply:
The LUH is currently being set at the same height at which the gov options are updated . So that while getting the exact key can be populated using LUH ,   using KEY+ LUH VALUE  , ( if LUH value is H+1 ,the keys would set using KEY+H,and it would try to get using KEY+(H+1) )